lua/entities/gmod_wire_expression2/core/custom/prop.lua

```diff
@@ -1473,25 +1479,57 @@ local function E2CollisionEventHandler()
 	processNextTick = false
 end
 
+local function startCollisionTracking(self,ent,entIndex,lambda)
+	local ctx = self
+	local callbackID = ent:AddCallback("PhysicsCollide",
+	function( us, cd )
+		table.insert(ctx.data.E2QueuedCollisions,{us=us,xcd=cd,cb=lambda})
+		if not processNextTick then
+			processNextTick = true
+			timer.Simple(0,E2CollisionEventHandler) -- A timer set to 0 runs next GM:Tick() hook
+		end
+	end)
+	self.data.E2TrackedCollisions[entIndex] = callbackID -- This ID is needed to remove the physcollide callback
+	ent:CallOnRemove("E2Chip_CCB" .. callbackID, function()
+		self.data.E2TrackedCollisions[entIndex] = nil
+	end)
+end
+
 e2function number trackCollision( entity ent )
+	-- If it's not registered, collisions will just stack up infinitely and not be flushed.
+	if not registered_chips[self.entity] then
+		self:forceThrow("event entityCollision(eexcd) is needed to use trackCollision(e)!")
+	end
 	if IsValid(ent) then
 		local entIndex = ent:EntIndex()
 		if self.data.E2TrackedCollisions[entIndex] then
-			return self:throw("Attempting to track collisions for an already tracked entity",0) -- Already being tracked.
+			return self:throw("Attempting to track collisions for an already tracked entity",0)
 		end
-		local ctx = self
-		local callbackID = ent:AddCallback("PhysicsCollide",
-		function( us, cd )
-			table.insert(ctx.data.E2QueuedCollisions,{us=us,xcd=cd})
-			if not processNextTick then
-				processNextTick = true
-				timer.Simple(0,E2CollisionEventHandler) -- A timer set to 0 runs next GM:Tick() hook
-			end
-		end)
-		self.data.E2TrackedCollisions[entIndex] = callbackID -- This ID is needed to remove the physcollide callback
-		ent:CallOnRemove("E2Chip_CCB" .. callbackID, function()
-			self.data.E2TrackedCollisions[entIndex] = nil
-		end)
+		startCollisionTracking(self,ent,entIndex)
+		return 1
+	end
+	return self:throw("Attempting to track collisions for an invalid entity",0)
+end
+
+e2function number trackCollision( entity ent, function cb )
```

Comment:
This is going to break glualint but it's valid & works ingame, fairly certain this may be the only vanilla e2 function(not automatically generated) that takes a lambda at present.